src/IdentityServer4.RavenDB.Storage/Services/CorsPolicyService.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using IdentityServer4.RavenDB.Storage.Entities;
+using IdentityServer4.Services;
+using Microsoft.Extensions.Logging;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Session;
+
+namespace IdentityServer4.RavenDB.Storage.Services
+{
+    public class CorsPolicyService : ICorsPolicyService
+    {
+        public CorsPolicyService(IAsyncDocumentSession session, ILogger<CorsPolicyService> logger)
+        {
+            Session = session ?? throw new ArgumentNullException(nameof(session));
+            Logger = logger;
+        }
+
+        protected IAsyncDocumentSession Session { get; }
+        protected ILogger<CorsPolicyService> Logger { get; }
+
+        /// <inheritdoc />
+        public async Task<bool> IsOriginAllowedAsync(string origin)
+        {
+            var query = Session.Query<Client>()
+                .Customize(x => x.WaitForNonStaleResults(TimeSpan.FromSeconds(5)))
```

Comment:
Why have these? It is better to wait on the save than the query.

Reply:
I agree, this is the original implementation strategy that I followed from other files and my suggestion is to use static indexes and wait for them after save, like you are saying I believe. I'll fix them.

Reply:
All waits have been removed from queries, now done as part of SaveChanges().